### PR TITLE
Piecewise gamma(CRV)

### PR DESCRIPTION
**Pull request by sidhu1012**
#### References to other Issues or PRs

Partial fix  #20031 

#### Brief description of what is fixed or changed
Fixed issue with gamma function in CRV.

#### Other comments
Added Piecewise to gamma.

#### Release Notes

- stats
       - Added changes to stats/crv_types.py file for Gamma.

### CODE REVIEW

File: sympy/stats/crv_types.py [hunk 1]
```diff
@@ -1684,7 +1684,8 @@ def check(k, theta):
 
     def pdf(self, x):
         k, theta = self.k, self.theta
-        return x**(k - 1) * exp(-x/theta) / (gamma(k)*theta**k)
+        expr = x**(k - 1) * exp(-x/theta) / (gamma(k)*theta**k)
+        return Piecewise((expr, self.set.as_relational(k)), (0, True))
```
Comment (Smit-create) [inline sympy/stats/crv_types.py:1688]:
```suggestion
        return Piecewise((expr, self.set.as_relational(x)), (0, True))
```

**Comment by Smit-create** [comment]
This should be changed in `crv_types.py` wherever necessary

**Comment by sidhu1012** [comment]
> This should be changed in `crv_types.py` wherever necessary

Okay

**Comment by sidhu1012** [comment]
@Smit-create  What's  _z  and from where to import it

**Comment by sidhu1012** [comment]
This issue will take tym as one change leads to cascade of changes

**Comment by czgdp1807** [comment]
Marking it stalled for now since, the broader question is what API should be used to fetch PMF of discrete and finite RVs.

**Comment by Upabjojr** [comment]
Please no `Piecewise` in `pdf`. There are other methods to deal with the support of RVs.

**Pull request closed without merging**